lib/backupds/datastore.go

```diff
@@ -146,11 +176,24 @@ func (d *Datastore) Sync(prefix datastore.Key) error {
 	return d.child.Sync(prefix)
 }
 
-func (d *Datastore) Close() error {
+func (d *Datastore) CloseLog() error {
 	d.backupLk.RLock()
 	defer d.backupLk.RUnlock()
 
-	return d.child.Close()
+	if d.closing != nil {
+		close(d.closing)
+		<-d.closed
+	}
+
+	return nil
+}
+
+func (d *Datastore) Close() error {
+	if err := d.child.Close(); err != nil {
+		return xerrors.Errorf("closing child datastore: %w", err)
+	}
+
+	return d.CloseLog()
```

Comment:
so here, things can go bad if we fail to close the `child`, because we will fail to close the log.
How about doing both in parallel and multierror combining the results?